packages/scalecube-services/src/services/ServiceCall.js

```diff
@@ -20,37 +23,44 @@ export class ServiceCall {
         this.microservices = ms;
     }
 
-    call(message: Message, type: "Observable" | "Promise") {
+    initialize(message: Message, type: "Observable" | "Promise") {
+        const getInstanceOfMessage = getInst(this.router);
+        if (!message) {
+            throw Error("Error: data was not provided");
+        }
         const chain$ = Observable
             .from([message])
             .map((message) => {
-                if (Array.isArray(message.data)) {
-                    return message;
+                if (!Array.isArray(message.data)) {
+                    throw Error("Message format error: data must be Array");
                 }
-                throw Error("Message format error: data must be Array");
-            })
-            .map(message => ({
-                message,
-                inst: this.router.route(message)
-            }))
-            .map((source) => {
-                if (source.inst && source.inst.service) {
-                    return source;
+                const inst = getInstanceOfMessage(message);
+                if (inst && inst.service) {
+                    return ({
+                        message,
+                        inst,
+                        thisMs: this.microservices,
+                        meta: getMeta(inst),
+                    });
                 }
                 throw Error(`Service not found error: ${message.serviceName}.${message.method}`);
             })
-            .map(source => ({
-                inst: source.inst,
-                message: source.message,
-                thisMs: this.microservices,
-                meta: source.inst.service.meta || source.inst.service.constructor.meta || {}
-            }))
             .pipe(source$ => this.microservices.preRequest(source$))
-            .map(obj => obj.inst)
+            .map(({ inst }) => inst)
             .switchMap(inst => utils.isLoader(inst) ?
                 Observable.from(new Promise(r => inst.service.promise.then(res => r(res)))) :
                 Observable.from([inst.service])
             )
+            .do((response) => {
```

Comment:
The idea that post request is mw
It need to run after response and you might modify it, so it can't be do